### PR TITLE
Added derivative of beta function and tests

### PR DESCRIPTION
**Pull request by nishnik**

### CODE REVIEW

File: symengine/derivative.cpp [hunk 1]
```diff
@@ -421,7 +420,16 @@ static RCP<const Basic> diff(const CLASS &self, \
...
         return self.diff_impl(x);
     }
-
+    static RCP<const Basic> diff(const Beta &self,
+            const RCP<const Symbol> &x) {
+        RCP<const Basic> beta_arg0 = self.get_args()[0];
+        RCP<const Basic> beta_arg1 = self.get_args()[1];
+        RCP<const Basic> diff_beta_arg0 = beta_arg0->diff(x);
+        RCP<const Basic> diff_beta_arg1 = beta_arg1->diff(x);
+        return mul(beta(beta_arg0, beta_arg1), add(mul(polygamma(zero, beta_arg0), diff_beta_arg0),
```
Comment (isuruf) [inline symengine/derivative.cpp]: `beta(beta_arg0, beta_arg1)` should just be `self.rcp_from_this()`

**Comment by nishnik** [comment]
Fixed @isuruf

**Comment by abhinavagarwal07** [comment]
:+1

**Comment by isuruf** [comment]
@nishnik, @abhinavagarwal07 thanks for the PR and the review.

**Pull request merged**